hera_cal/reflections.py

```diff
@@ -847,10 +861,22 @@ def interp_u(self, umodes, times, full_times=None, uflags=None, keys=None, overw
             # setup GP
             GP = gp.GaussianProcessRegressor(kernel=kernel, optimizer=optimizer, normalize_y=False)
 
+            # setup regression data
+            X = times - Xmean
+            Y = umodes[k][:, :Ninterp].copy()
+            F = uflags[k].copy()
+
+            # mirror if desired
+            if Nmirror > 0:
+                assert Nmirror < X.size, "Nmirror can't be equal or larger than Ntimes"
+                _, X = vis_clean.zeropad_array(Y, binvals=X, zeropad=Nmirror, axis=0)
+                Y = np.concatenate([Y[1:Nmirror + 1, :][::-1, :], Y, Y[-Nmirror - 1:-1, :][::-1, :]], axis=0)
+                F = np.concatenate([F[1:Nmirror + 1, :][::-1, :], F, F[-Nmirror - 1:-1, :][::-1, :]], axis=0)
+
             # setup regression data: get unflagged data
-            select = ~np.max(uflags[k], axis=1)
-            X = times[select, None] - Xmean
-            Y = umodes[k][select, :].copy()
+            select = ~np.max(F, axis=1)
```

Comment:
I really don't like using max's and mins for comparing booleans. Instead, I'd use np.any or np.all.

Reply:
ok